pootle/static/js/editor/app.js

```diff
@@ -1262,55 +1264,60 @@ PTL.editor = {
     );
 
-    if (!canShow) {
-      return { leadIn, leadOut };
-    }
-
-    if (this.units.begin === 0) {
-      leadIn.push(nt(
-        'Found %(total)s unit.',
-        'Found %(total)s units.',
-        this.units.total, { total: this.units.total }
-      ));
-    } else {
-      // there are some units above;
-      // this can only happen when filter is set to 'all'
-      leadIn.push(nt(
-        'There are %(count)s more unit above.',
-        'There are %(count)s more units above.',
-        this.units.total, { count: this.units.begin }
-      ));
+    if (canShow) {
+      if (this.units.begin === 0) {
+        leadIn.push(nt(
+          'Found %(total)s unit.',
+          'Found %(total)s units.',
+          this.units.total, { total: this.units.total }
+        ));
+      } else {
+        // there are some units above;
+        // this can only happen when filter is set to 'all'
+        leadIn.push(nt(
+          'There are %(count)s more unit above.',
+          'There are %(count)s more units above.',
+          this.units.total, { count: this.units.begin }
+        ));
 
-      leadIn.push(t('Reload the page to see them.'));
+        leadIn.push(t('Reload the page to see them.'));
+      }
     }
 
     if (this.units.end === this.units.total) {
-      leadOut.push(t('Congratulations, you went through all units!'));
+      if (showCongratulationsMessage) {
+        leadOut.push(t('Congratulations, you went through all units!'));
+        leadOut.push($('.back-to-browser').html());
+      }
     } else {
-      leadOut.push(t('Congratulations, you went through a good number of units!'));
+      if (showCongratulationsMessage) {
+        leadOut.push(t('Congratulations, you went through a good number of units!'));
+      }
 
-      if (this.filter === 'all') {
-        leadOut.push(t('Reload the page to see some more.'));
-      } else if (this.filter === 'search') {
-        leadOut.push(t(
-          `There were more units that matched your search criteria.
+      if (canShow) {
+        if (this.filter === 'all') {
+          leadOut.push(t('Reload the page to see more units.'));
+        } else if (this.filter === 'search') {
+          leadOut.push(t(
+            `There were more units that matched your search criteria.
           Please refine your search.`
-        ));
-      } else {
-        leadOut.push(t(
-          `There were more units that matched your filter.
+          ));
+        } else {
+          leadOut.push(t(
+            `There were more units that matched your filter.
           Reload the page to see if there is more stuff to go through,
           or use search to find specific units.`
-        ));
+          ));
+        }
       }
     }
 
     return { leadIn, leadOut };
   },
 
   /* redraws the translate table view rows */
-  redrawViewRows() {
-    const messages = this.populateLeadInOutMessages();
+  redrawViewRows(showCongratulationsMessage) {
```

Comment:
Like we do in other parts of the code, I suggest not to use naked boolean values as parameters ([to avoid the boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap)) and rather use an object so the call remains explicit: `redrawViewRows({ showCongratulationsMessage: true })`. The signature can be changed to `redrawViewRows({ showContratulationsMessage = false })` (also providing a default value) so the function body wouldn't require any changes.